sdk/core/core/src/az_client_secret_credential.c

```diff
@@ -177,9 +180,31 @@ AZ_INLINE AZ_NODISCARD az_result az_token_credential_update(
 
   {
     az_json_value value;
-    AZ_RETURN_IF_FAILED(az_json_get_object_member(body, AZ_STR("access_token"), &value));
+
+    clock_t lifetime_seconds = 0;
+    if (requested_at > 0) {
+      az_span expires_in = { 0 };
+      if (az_succeeded(az_json_get_object_member(body, AZ_STR("expires_in"), &value))
+          && az_succeeded(az_json_value_get_string(&value, &expires_in))) {
+        clock_t number = 0;
+        for (size_t i = 0; i < expires_in.size; ++i) {
+          uint8_t const c = expires_in.begin[i];
+          if (c >= '0' && c <= '9') {
+            number = number * 10 + (c - '0');
+          } else {
+            number = 0;
+            break;
+          }
+        }
+
+        if (number > 0) {
+          lifetime_seconds = (number - (3 * 60)) * CLOCKS_PER_SEC;
```

Comment:
maybe add comments about what 3*60 is doing..  3 seconds?  3 minutes?  why 3?

Reply:
I changed naming, I hope it i clearer now. 3 minutes before expiration is an arbitrary number, at least for now. Later we may want to get it from options.